paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -285,13 +285,14 @@ def sync_horizontal_pod_autoscaler(self, kube_client: KubeClient) -> None:
         In order for autoscaling to work, there needs to be at least two configurations
         min_instnace, max_instance, and there cannot be instance.
         """
-        self.logging.info(
-            f"Syncing HPA setting for {self.item.metadata.name}/name in {self.item.metadata.namespace}"
-        )
+        should_have_hpa = self.should_have_hpa()
```

Comment:
What if instead of this we just moved the logic in `should_have_hpa` into `get_autoscaling_metric_spec` and just had that method return `None` if there shouldn't be an hpa?  I think part of why this bug exists is that this logic is split across two modules, meaning that it's possible to call the canonical "create me an HPA" method for Deployments that shouldn't have HPAs and have it return something.

All the logic that makes up `should_have_hpa` is just method calls on `self.soa_config` anyway.

Reply:
OK, I've tried to streamline the logic - let me know what you think.